template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/core/CoreConfigProvider.java

```diff
@@ -59,39 +59,45 @@ public class CoreConfigProvider extends AbstractRoleConfigProvider {
 
     @Override
     protected List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, TemplatePreparationObject source) {
-        if (entitlementService.isSDXOptimizedConfigurationEnabled(ThreadBasedUserCrnProvider.getAccountId())
-            && source.getStackType().equals(StackType.DATALAKE)) {
-            return List.of(config(HADOOP_RPC_PROTECTION, "privacy"));
-        }
         return List.of();
     }
 
     @Override
     public List<ApiClusterTemplateConfig> getServiceConfigs(CmTemplateProcessor templateProcessor, TemplatePreparationObject source) {
         List<ApiClusterTemplateConfig> apiClusterTemplateConfigs = new ArrayList<>();
-        Optional<ApiClusterTemplateConfig> roleConfig = templateProcessor.getRoleConfig(CORE_SETTINGS, STORAGEOPERATIONS, CORE_DEFAULTFS);
-        if (roleConfig.isEmpty()) {
-            ConfigUtils.getStorageLocationForServiceProperty(source, CORE_DEFAULTFS)
-                    .ifPresent(location -> apiClusterTemplateConfigs.add(config(CORE_DEFAULTFS, location.getValue())));
+        if (!templateProcessor.isRoleTypePresentInService(HDFS, Lists.newArrayList(NAMENODE))
```

Comment:
I don't know enough to answer this myself, but I always get concerned when we add a new conditional around central logic. 

What happens to calls to this method that don't satisfy this condition? Do they get an empty config list? How does that differ from what we're doing right now?

My main concern is that we accidentally give empty configs to other services which need to go through the rest of this method.